examples/burger-chef.jl

```diff
@@ -4,26 +4,50 @@ using DiscSimulations
 using Plots
 using DifferentialEquations
 using Printf
+using Trixi
 
 N = 512
 xmax = 2π
 
-simple_x, _, simple_problem = BurgerSimple.setup(N, xmax)
+function initial_condition_burgers(x)
+    u = 1.0
+    if(x[1] >= 0.333 && x[1] <= 0.666)
+        u = u + 0.5*sin(2.0*π*(x[1]-0.333)/0.333)
+    end
+    return u
+end
+
+burgerInit(x) = @. 1 - cos(x)
+
+simple_x, _, simple_problem = BurgerSimple.setup(N, xmax, initial_condition_burgers)
 simple_sol =
     @time solve(simple_problem, Rodas5(autodiff = false); reltol = 1e-7, abstol = 1e-7)
 
-spectral_x, _, T, Ti, spectral_problem = BurgerSpectral.setup(N, xmax)
+spectral_x, _, T, Ti, spectral_problem = BurgerSpectral.setup(N, xmax, initial_condition_burgers)
```

Comment:
You need to create a wrapper function that makes the initial condition function formats compatible. `BurgerSimple` and `BurgerSpectral` both expect functions in the form of:
```julia
f(::AbstractArray)::AbstractArray
```
whereas your function is of the form
```julia
f(::AbstractArray)::Number
```

To fix, define something like
```julia
function _wrap_init_conditions(f::Function)
    function _wrapper(x)
        map(x) do i
            f(SVector(i))
        end
    end
end
```

You can then use this as 
```julia
... = BurgerSpectral.setup(N, xmax, _wrap_init_conditions(initial_condition_burgers))
```